spectrocloud/resource_cluster_libvirt.go

```diff
@@ -0,0 +1,643 @@
+package spectrocloud
+
+import (
+	"context"
+	"log"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/spectrocloud/gomi/pkg/ptr"
+	"github.com/spectrocloud/hapi/models"
+	"github.com/spectrocloud/terraform-provider-spectrocloud/pkg/client"
+)
+
+func resourceClusterLibvirt() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceClusterVirtCreate,
+		ReadContext:   resourceClusterLibvirtRead,
+		UpdateContext: resourceClusterVirtUpdate,
+		DeleteContext: resourceClusterDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		SchemaVersion: 2,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"tags": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Set:      schema.HashString,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"cluster_profile_id": {
```

Comment:
@nikchern We can remove cluster_profile_id and pack

Reply:
@rishi-anand I removed it.